src/dmqnode/storage/engine/overflow/file/IndexFile.d

```diff
@@ -237,16 +237,18 @@ class IndexFile: PosixFile
     {
         this.reset();
 
-        this.fmt_io_signal_blocker.callBlocked({
-            iterate((cstring name, ChannelMetadata channel)
-            {
-                int n = fprintf(this.stream, "%.*s %lu %llu %lld %lld\n".ptr,
-                                name.length, name.ptr,
-                                channel.records, channel.bytes,
-                                channel.first_offset, channel.last_offset);
+        scope on_channel = (cstring name, ChannelMetadata channel)
+        {
+            int n = fprintf(this.stream, "%.*s %lu %llu %lld %lld\n".ptr,
+                            name.length, name.ptr,
+                            channel.records, channel.bytes,
+                            channel.first_offset, channel.last_offset);
 
-                this.enforce(n >= 0, "error writing index");
-            });
+            this.enforce(n >= 0, "error writing index");
+        };
+
+        this.fmt_io_signal_blocker.callBlocked({
+            iterate(on_channel);
```

Comment:
Shouldn't we change the function to accept `void delegate(/* d1to2fix_inject: scope */` ...)` instead ?
IIUC the issue is that `d1to2fix` does not add scope in parameter list of delegates.